src/api/app/mixins/webui/manage_relationships.rb

```diff
@@ -1,50 +1,48 @@
 module Webui::ManageRelationships
-  def load_obj
-    login = params[:userid]
-    return User.find_by_login!(login) if login
-    title = params[:groupid]
-    return ::Group.find_by_title!(title) if title
-    raise MissingParameterError, 'Neither user nor group given'
+  def redirect_after_save(what)
+    if switch_to_webui2
+      redirect_to(users_path)
+    elsif what == :person
+      redirect_to(add_path(:add_person))
+    else
+      redirect_to(add_path(:add_group))
+    end
   end
 
-  def save_person
+  def save_person_or_group(what)
     authorize main_object, :update?
     begin
-      Relationship.add_user(main_object, load_obj, Role.find_by_title!(params[:role]), nil, true) # report error on duplicate
+      Relationship::AddRole.new(main_object, Role.find_by_title!(params[:role]), check: true, user: params[:userid], group: params[:groupid]).add_role # report error on duplicate
       main_object.store
     rescue NotFoundError,
-           Relationship::SaveError => e
+           Relationship::AddRole::SaveError => e
       flash[:error] = e.to_s
-      switch_to_webui2 ? redirect_to(users_path) : redirect_to(add_path(:add_person))
-      return
+      return redirect_after_save(what)
     end
     respond_to do |format|
       format.js { render json: {}, status: :ok }
       format.html do
-        flash[:success] = "Added user #{params[:userid]} with role #{params[:role]}"
+        success_str = what == :person ? "user #{params[:userid]}" : "group #{params[:groupid]}"
+        flash[:success] = "Added #{success_str} with role #{params[:role]}"
         redirect_to users_path
       end
     end
   end
 
+  def save_person
+    save_person_or_group(:person)
+  end
+
   def save_group
-    authorize main_object, :update?
-    begin
-      Relationship.add_group(main_object, load_obj, Role.find_by_title!(params[:role]), nil, true) # report error on duplicate
-      main_object.store
-    rescue NotFoundError,
-           Relationship::SaveError => e
-      flash[:error] = e.to_s
-      switch_to_webui2 ? redirect_to(users_path) : redirect_to(add_path(:add_group))
-      return
-    end
-    respond_to do |format|
-      format.js { render json: {}, status: :ok }
-      format.html do
-        flash[:success] = "Added group #{params[:groupid]} with role #{params[:role]}"
-        redirect_to users_path
-      end
-    end
+    save_person_or_group(:group)
+  end
+
+  def load_obj
```

Comment:
```load_obj``` is a real bad name.. if you are already making the code better, could you please rename it too?

Reply:
actually I only moved it to private - it's only used in another function.

Reply:
yes I know it, but to rename it would be great too.